docs/CONTRIBUTING.md

```diff
@@ -94,6 +94,146 @@ Now force push to update your PR:
 git push --force
 ```
 
+## API tests
+
+There are two kinds of tests in the API: unit (`phpunit`) and integration tests (`behat`).
+
+Both `phpunit` and `behat` are development dependencies and should be available in the `vendor` directory.
+
+API tests must be run in the `api` directory.
+
+### PHPUnit and coverage generation
+
+To launch unit tests:
+
+```bash
+bin/phpunit
+```
+
+If you want coverage, you will need the `phpdbg` package and run:
+
+```bash
+phpdbg -qrr bin/phpunit --coverage-html coverage
+```
+
+Sometimes there might be an error with too many open files when generating coverage. To fix this, you can increase the
+`ulimit`, for example:
+
+```bash
+ulimit -n 4000
```

Comment:
A better fix (that will speed up the build) is to set the `api_platform.metadata_cache` container's parameter to true in `test` env.
We should update the Flex recipe, `api-platform/api-platform` and the demo to set it by default.